driver-core/src/main/java/com/datastax/driver/core/policies/FallthroughRetryPolicy.java

```diff
@@ -88,6 +88,19 @@ public RetryDecision onUnavailable(Statement statement, ConsistencyLevel cl, int
         return RetryDecision.rethrow();
     }
 
+    /**
+     * {@inheritDoc}
+     * <p>
+     * This implementation triggers a retry on the next host in the query plan,
+     * regardless of the consistency level, the number of retries or
+     * the possibility that the mutation has been applied server-side.
+     */
+    @Override
+    public RetryDecision onUnexpectedError(Statement statement, ConsistencyLevel cl, int nbRetry) {
+        return RetryDecision.rethrow();
```

Comment:
I am modifying the decision here as compared to the 2.1 branch, which returns `tryNextHost(cl)` to mimic the driver's behavior prior to JAVA-819.